### PR TITLE
Add Safari versions for Document API

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
This PR adds real values for Safari (Desktop and iOS/iPadOS) for the `Document` API.  The data was derived from their handlers (AKA `on*`).

### CODE REVIEW

File: api/Document.json [hunk 1]
```diff
@@ -5425,10 +5425,12 @@
               "version_added": "32"
             },
             "safari": {
-              "version_added": null
+              "version_added": "5.1",
```
Comment (foolip) [inline api/Document.json:5428]: This doesn't match https://developer.mozilla.org/en-US/docs/Web/API/Element/requestFullScreen#browser_compatibility, the event can't have been fired before the API was supported.

File: api/Document.json [hunk 2]
```diff
@@ -8786,10 +8790,10 @@
               "version_added": null
             },
             "safari": {
-              "version_added": null
+              "version_added": "10.1"
```
Comment (foolip) [inline api/Document.json:8793]: https://developer.mozilla.org/en-US/docs/Web/API/Pointer_Lock_API#browser_compatibility says 10.1 for the unprefixed API, but also that there was a webkit-prefixed one. Is that right, and did that fire a prefixed event that should be added here?

File: api/Document.json [hunk 3]
```diff
@@ -8864,10 +8868,10 @@
...
```
Comment (foolip) [inline api/Document.json:8871]: Ditto.

**Review by foolip — CHANGES_REQUESTED** [review]

File: api/Document.json (hunk 1 above)
Comment (vinyldarkscratch) [reply api/Document.json:5428]: It appears that `api.Element.requestFullScreen` is incorrect, then.  I just launched Safari 5.1 on BrowserStack and ran `document.forms[0].webkitRequestFullScreen`, and the result came up as a function.  I'll submit a PR to correct it.

File: api/Document.json (hunk 2 above)
Comment (vinyldarkscratch) [reply api/Document.json:8793]: I took a closer look into this one, and found that the data we have for `api.Element.requestPointerLock` is inaccurate.  There was no prefixed version for the function before Safari 10.1, or even in 10.1 itself.  Tested in SauceLabs in Safari 10.0, 10.1, and 9.0.

I'll write a PR to fix up its data.

File: api/Document.json (hunk 1 above)
Comment (foolip) [reply api/Document.json:5428]: Thanks! That's https://github.com/mdn/browser-compat-data/pull/11162, would you mind folding those changes into that PR for easier review as a group?

File: api/Document.json (hunk 2 above)
Comment (foolip) [reply api/Document.json:8793]: Would you mind folding these changes into https://github.com/mdn/browser-compat-data/pull/11164 for easier review?

File: api/Document.json (hunk 1 above)
Comment (vinyldarkscratch) [reply api/Document.json:5428]: I've merged the changes from this PR into that one!  This PR is now no longer needed.

**Pull request closed without merging**